src/charts/legend.js

```diff
@@ -202,16 +204,53 @@ define(function(require){
          * @return {void}
          * @private
          */
-        function centerLegendOnSVG() {
+        function centerInlineLegendOnSVG() {
             let legendGroupSize = svg.select('g.legend-container-group').node().getBoundingClientRect().width + getLineElementMargin();
             let emptySpace = width - legendGroupSize;
+            let newXPosition = (emptySpace/2);
 
             if (emptySpace > 0) {
                 svg.select('g.legend-container-group')
-                    .attr('transform', `translate(${emptySpace/2},0)`)
+                    .attr('transform', `translate(${newXPosition},0)`)
             }
         }
 
+        /**
+         * Centers the legend on the chart given that is a stack of labels
+         * @return {void}
+         * @private
+         */
+        function centerVerticalLegendOnSVG() {
+            let legendGroupSize = svg.select('g.legend-container-group').node().getBoundingClientRect().width;
+            let emptySpace = width - legendGroupSize;
+            let newXPosition = (emptySpace / 2) - (legendGroupSize / 2);
+
+            if (emptySpace > 0) {
+                svg.select('g.legend-container-group')
+                    .attr('transform', `translate(${newXPosition},0)`)
+            }
+        }
+
+        /**
+         * Makes sure the types of the data are right and checks if it has quantities
+         * @param {LegendChartData} data
+         * @private
+         */
+        function cleanData(data) {
+            hasQuantities = data.filter(hasQuantity).length === data.length;
+
+            return data
+                .reduce((acc, d) => {
+                    if (d.quantity !== undefined) {
+                        d.quantity = +d.quantity;
+                    }
+                    d.name = String(d.name);
+                    d.id = +d.id;
```

Comment:
for this one as well: not sure what the `+` sign in front of the variable is supposed to do